lib/web/app/handler.go

```diff
@@ -567,7 +567,7 @@ func makeAppRedirectURL(r *http.Request, proxyPublicAddr, hostname string) strin
 	// as a query part. Append query that was originally part of the URL.
 	query := fmt.Sprintf("path=%s", url.QueryEscape(r.URL.Path))
```

Comment:
This part with `url.QueryEscape(r.URL.Path)` is quite nuanced but we have no comment about it.

```go
package main

import (
	"fmt"
	"log"
	"net/url"
)

func main() {
	spacesURL, err := url.Parse("https://simplehttpserver.teleport-local.dev:3090/siema%20elo/?foo=bar%20baz&query=elo")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(spacesURL.Path)
	fmt.Println(spacesURL.EscapedPath())
	fmt.Println(url.QueryEscape(spacesURL.Path))
	fmt.Println(url.QueryUnescape(url.QueryEscape(spacesURL.Path)))
}
```

The output:

```
/siema elo/
/siema%20elo/
%2Fsiema+elo%2F
/siema elo/ <nil>
```

Could you add a comment here? It could explain that `URL.Path` is the decoded form of path, where any urlencoded parts or decoded. Spaces are encoded in query params as `+` whereas in path they're encoded as `%20`. Since we want to put the path into the query params, we need to take the decoded path and encode it use in query params with `url.QueryEscape`.

This is probably intuitive for anyone with extensive experience with Go's net/url, but understanding that took me a lot of jumping around different definitions in the docs, especially because of the nonstandard thing we do (putting the path into the query params).